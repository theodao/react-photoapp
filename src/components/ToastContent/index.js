/* eslint-disable indent */
import React from 'react';
import { Icon } from '@gotitinc/design-system';

export default ({
  iconName = 'helpCircle',
  iconSize = 'medium',
  content = '',
  title = 'Default',
}) => (
  <div className="u-flex u-flexGrow-1">
    <div className="u-marginRightExtraSmall">
      <Icon name={iconName} size={iconSize} />
    </div>
    <div className="u-flexGrow-1">
      <div className="u-fontMedium u-marginBottomExtraSmall">{title}</div>
      <div>
        {typeof content === 'string'
          ? content
          : content.map((item) => {
              return <li>{item}</li>;
            })}
      </div>
    </div>
  </div>
);
