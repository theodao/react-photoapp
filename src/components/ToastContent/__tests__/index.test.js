import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import ToastContent from '../';

describe('Testing toast content component', () => {
  afterEach(cleanup);

  it('Should match snapshot', () => {
    const { asFragment } = render(<ToastContent />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('If content was an array it should be rendered', () => {
    const wrapper = shallow(<ToastContent content={['a']} />);

    console.log(wrapper.debug());
  });
});
