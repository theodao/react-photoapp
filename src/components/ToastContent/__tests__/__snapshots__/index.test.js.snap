// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing toast content component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="u-flex u-flexGrow-1"
  >
    <div
      class="u-marginRightExtraSmall"
    >
      <svg
        aria-hidden="true"
        class=""
        focusable="false"
        height="32px"
        style="display: inline-block; vertical-align: middle;"
        viewBox="0 0 24 24"
        width="32px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,3 C7.02836538,3 3,7.02836538 3,12 C3,16.9716346 7.02836538,21 12,21 C16.9716346,21 21,16.9716346 21,12 C21,7.02836538 16.9716346,3 12,3 Z M11.8139423,16.1538462 C11.3033654,16.1538462 10.8879808,15.7644231 10.8879808,15.2625 C10.8879808,14.7649038 11.3033654,14.3711538 11.8139423,14.3711538 C12.3288462,14.3711538 12.7442308,14.7605769 12.7442308,15.2625 C12.7442308,15.7644231 12.3331731,16.1538462 11.8139423,16.1538462 Z M13.5533654,11.9610577 C12.8004808,12.3980769 12.5451923,12.7182692 12.5451923,13.2721154 L12.5451923,13.6139423 L11.04375,13.6139423 L11.0307692,13.2418269 C10.9572115,12.3504808 11.26875,11.7966346 12.0519231,11.3379808 C12.7831731,10.9009615 13.0903846,10.6240385 13.0903846,10.0875 C13.0903846,9.55096154 12.5711538,9.15721154 11.9264423,9.15721154 C11.2730769,9.15721154 10.8014423,9.58125 10.7668269,10.2216346 L9.23076923,10.2216346 C9.26105769,8.82836538 10.2908654,7.84182692 12.0302885,7.84182692 C13.6528846,7.84182692 14.7692308,8.74182692 14.7692308,10.0355769 C14.7692308,10.8966346 14.3538462,11.4894231 13.5533654,11.9610577 L13.5533654,11.9610577 Z"
          style="fill: currentColor;"
        />
      </svg>
    </div>
    <div
      class="u-flexGrow-1"
    >
      <div
        class="u-fontMedium u-marginBottomExtraSmall"
      >
        Default
      </div>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
